### PR TITLE
Mark of the Falling Tower meteor costs are now configurable

### PR DESCRIPTION
**Pull request by keraldi**
via an additonal field "cost" in the .json config files

 if BM is updated from an old version, this would require meteor costs to be added manually for custom meteors; meteors would be for free (LP-wise) otherwise.

 These are the new default costs:
 IRON: 1,000,000 (same as before)
 GOLD: 2,500,000
 DIAMOND: 5,000,000 (requires Archmage Blood Orb)

Default Meteor versions have been increased by 1 in order to update their configuration.

### CODE REVIEW

**Comment by keraldi** [comment]
#1475

File: src/main/java/WayofTime/bloodmagic/item/ItemBoundTool.java [hunk 1]
```diff
@@ -244,7 +244,7 @@ protected void sharedHarvest(ItemStack stack, World world, EntityPlayer player,
                 if (silkTouch && blockState.getBlock().canSilkHarvest(world, blockPos, world.getBlockState(blockPos), player))
                     ItemHandlerHelper.giveItemToPlayer(player, new ItemStack(Item.getItemFromBlock(blockState.getBlock())));
                 else {
-                    NonNullList<ItemStack> itemDrops = new NonNullList<>();
+                    NonNullList<ItemStack> itemDrops = NonNullList.create();
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/item/ItemBoundTool.java]: Irrelevant change I'll do this myself.

**Comment by noobanidus** [comment]
ilu

File: src/main/java/WayofTime/bloodmagic/meteor/Meteor.java [hunk 2]
```diff
@@ -29,6 +30,7 @@ public Meteor(ItemStack catalystStack, List<MeteorComponent> components, float e
         for (MeteorComponent component : components)
             weight += component.getWeight();
         this.maxWeight = weight;
+        this.cost = 1000000;
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/meteor/Meteor.java]: GSON initializes classes with reflection and not the constructor. The default definition should be set at the field declaration.

**Review by TehNut — CHANGES_REQUESTED** [review]

**Review by TehNut — APPROVED** [review]

**Pull request merged**